src/plugins/topic_viewer/TopicViewer.cc

```diff
@@ -0,0 +1,330 @@
+#define PLOT_KEY "plottable"
+
+#define NAME_ROLE 51
+#define TYPE_ROLE 52
+#define TOPIC_ROLE 53
+#define PATH_ROLE 54
+#define PLOT_ROLE 55
+
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+  class TopicsModel : public QStandardItemModel
+  {
+    /// \brief roles and names of the model
+    public: QHash<int, QByteArray> roleNames() const override
+    {
+      QHash<int, QByteArray> roles;
+      roles[NAME_ROLE] = NAME_KEY;
+      roles[TYPE_ROLE] = TYPE_KEY;
+      roles[TOPIC_ROLE] = TOPIC_KEY;
+      roles[PATH_ROLE] = PATH_KEY;
+      roles[PLOT_ROLE] = PLOT_KEY;
+      return roles;
+    }
+  };
+
+  class TopicViewerPrivate
+  {
+    /// \brief Node for Commincation
+    public: ignition::transport::Node node;
+
+    /// \brief Model to create it from the available topics and messages
+    public: TopicsModel *model;
+  };
+}
+}
+}
+using namespace ignition;
+using namespace gui;
+using namespace plugins;
+
+TopicViewer::TopicViewer() :Plugin(), dataPtr(new TopicViewerPrivate)
+{
+  using namespace google::protobuf;
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_DOUBLE);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_FLOAT);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_INT32);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_INT64);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_UINT32);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_UINT64);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_BOOL);
+
+  this->plottingMode = false;
+
+  this->CreateModel();
+
+  ignition::gui::App()->Engine()->rootContext()->setContextProperty(
+                "TopicsModel", this->dataPtr->model);
+}
+
+//////////////////////////////////////////////////
+TopicViewer::~TopicViewer()
+{
+}
+
+//////////////////////////////////////////////////
+void TopicViewer::LoadConfig(const tinyxml2::XMLElement *)
+{
+  if (this->title.empty())
+    this->title = "Topic Viewer";
+}
+
+//////////////////////////////////////////////////
+void TopicViewer::CreateModel()
+{
+  this->dataPtr->model = new TopicsModel();
+
+  std::vector<std::string> topics;
+  this->dataPtr->node.TopicList(topics);
+
+  for (unsigned int i = 0; i < topics.size(); ++i)
+  {
+    std::vector<ignition::transport::MessagePublisher> infoMsgs;
+    this->dataPtr->node.TopicInfo(topics[i], infoMsgs);
+    std::string msgType = infoMsgs[0].MsgTypeName();
+    this->AddTopic(topics[i], msgType);
+  }
+}
+
+//////////////////////////////////////////////////
+void TopicViewer::SetPlottingMode(bool _mode)
+{
+  this->plottingMode = _mode;
+}
+
+//////////////////////////////////////////////////
+bool TopicViewer::PlottingMode()
+{
+  return this->plottingMode;
+}
+
+//////////////////////////////////////////////////
+void TopicViewer::AddTopic(const std::string &_topic,
+                           const std::string &_msg)
+{
+  // remove 'ignition.msgs.' from msg name
```

Comment:
Is this really necessary? It's possible to publish messages that are not `ignition.msgs`, so it may be helpful for the user to see the full type name.